app/models/assets/import.rb

```diff
@@ -337,40 +311,35 @@ def annotate_wells(asset, remote_asset, fact_changes)
       end
     end
 
-    def _annotate_wells(asset, remote_asset, fact_changes)
+    def annotate_tubes(asset, remote_asset, fact_changes)
       fact_changes.tap do |updates|
-        if remote_asset.try(:wells, nil)
-          remote_asset.wells.each do |well|
-            local_well = Asset.find_or_create_by!(:uuid => well.uuid)
-            if (well.try(:aliquots, nil)&.first&.sample&.supplier&.sample_name)
-              updates.replace_remote(asset, 'contains', local_well)
+        if remote_asset.try(:racked_tubes)
+          remote_asset.racked_tubes.each do |racked_tube|
+            remote_tube = racked_tube.tube
+            local_tube = Asset.find_or_create_by!(:uuid => remote_tube.uuid)
```

Comment:
We can import the barcode of the remote tube as well

Reply:
Yes, thanks, I realised I hadn't done that yesterday. I checked with Laura and she said it's not a blocker for them, so I'll merge this in and add it as a new task in the backlog.

Reply:
Actually it looked like it would be pretty quick, so I've done it here - https://github.com/sanger/samples_extraction/pull/108/commits/2a8ccbb4798d0d682afd5744e4d3237e09698f6e
Thanks